src/widget/wsearchlineedit.cpp

```diff
@@ -277,7 +277,17 @@ void WSearchLineEdit::updateText(const QString& text) {
     if (isEnabled() && (m_state == State::Active)) {
         updateClearButton(text);
         DEBUG_ASSERT(m_debouncingTimer.isSingleShot());
-        m_debouncingTimer.start(s_debouncingTimeoutMillis);
+        if (s_debouncingTimeoutMillis > 0) {
+            m_debouncingTimer.start(s_debouncingTimeoutMillis);
+        } else {
+            // Deactivate the timer if the timeout is invalid.
+            // Disabling the timer permanently by setting the timeout
+            // to an invalid value is an expected and valid use case.
+            if (m_debouncingTimer.isActive()) {
+                kLogger.debug() << "Deactivating debouncing timer";
```

Comment:
Currently the lower bound checked by the preferences UI is 100 ms and we should **never** enter this branch. The branch will be entered **at most once** after encountering a timeout value <= 0 ms, because QTimer::isActive() will return false once the timer has been stopped and is never started again.

Reply:
I noticed that you are not guaranteed to see the log message at least once if the timeout is invalid. Ok, maybe the log doesn't help here.